Feature/add tls support to connections

omarmostafa opened the pull request.

Commit:
[TLS/Support]
- Add TLS support to tcp connection

Commit:
[TLS/Support]
- Add public and secret key files

Commit:
update readme file to replace telnet with openssl

xfoxawy commented on .idea/.gitignore:
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
+/workspace.xml
```
ignore this file

xfoxawy commented on .idea/go-in-memory-db.iml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
ignore this file

xfoxawy commented on .idea/modules.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
.gitignore .idea

xfoxawy commented on server.crt:
```diff
@@ -0,0 +1,17 @@
+-----BEGIN CERTIFICATE-----
```
remove the keys and and add in read me how to generate them

xfoxawy commented on go-in-memory-db.go:
```diff
@@ -13,7 +14,13 @@ var addr = ":6380"
 
 	go log.Printf("started server at %s", addr)
-	err := redcon.ListenAndServe(addr,
+	cer, err := tls.LoadX509KeyPair("server.crt", "server.key")
```
lets do a flag based init.
for example `./run --ssl=true --private-key=server.crt --public-key=server.keey`
other wise start it without ssl

xfoxawy commented on readme.md:
````diff
@@ -47,7 +47,7 @@ go test -cover ./...
 then through Telnet or any TCP connection
 ```
-telnet 127.0.0.1 8080
+openssl s_client -starttls smtp -connect 127.0.0.1:8080
````
we should document based on `redis-cli`
this document can give u more insights 
https://redislabs.com/blog/stunnel-secure-redis-ssl/

Commit:
[TLS/Support]
- Add tls as a flag in command arguments and generate private and public key in readme

Commit:
[TLS/Support]
- ignore .idea

Commit:
remove .idea

Commit:
remove public and private keys
